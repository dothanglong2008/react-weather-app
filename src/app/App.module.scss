* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html {
  font-size: calc(10px + 1vmin);
}

ul {
  list-style: none;
}

img,
picture,
figure {
  width: 100%;
  object-fit: none;
}

button,
a {
  cursor: pointer;
}

.app {
  background-color: #ebebeb;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}
