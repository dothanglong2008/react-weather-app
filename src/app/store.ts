import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import weatherReducer from "../features/weather/weatherSlice";
import searchFormReducer from "../components/weather/search/searchFormSlice";

export const store = configureStore({
  reducer: {
    weather: weatherReducer,
    searchForm: searchFormReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
