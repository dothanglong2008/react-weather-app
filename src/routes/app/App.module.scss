$background-color-primary: #ffffff;
$text-color-primary: #000000;
$font-size-primary: 16px;

* {
  margin: 0;
  padding: 0;
  width: 100%;
}

body {
  background-color: $background-color-primary;
  color: $text-color-primary;
  font-size: $font-size-primary;
}

ul {
  list-style: none;
}

img,
picture,
video {
  max-width: 100%;
}
