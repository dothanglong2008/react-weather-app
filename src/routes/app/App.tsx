import Item from "../../components/weather/single-item/item/Item";
import styles from "./App.module.scss";

function App() {
  return (
    <div className={styles.app}>
      <Item />
    </div>
  );
}

export default App;
