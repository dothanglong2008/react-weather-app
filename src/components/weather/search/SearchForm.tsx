import styles from "./SearchForm.module.scss";
import { FaSearch } from "react-icons/fa";
import { useAppDispatch, useAppSelector } from "../../../app/hooks";
import { changeLocation } from "../../../features/weather/weatherSlice";
import { selectLocationField } from "./searchFormSlice";
import { changeLocationField } from "./searchFormSlice";
import useResetForm from "../../../hooks/useResetForm";
import LocationSuggestion from "../location-suggestion/LocationSuggestion";

function SearchForm() {
  const locationField = useAppSelector(selectLocationField);
  const dispatch = useAppDispatch();
  useResetForm();
  return (
    <form
      name="search"
      className={styles.form}
      onSubmit={(event) => {
        event.preventDefault();
        dispatch(changeLocation(locationField));
      }}
    >
      <div className={styles.searchContainer}>
        <div className={styles.inputContainer}>
          <input
            id="location"
            type="text"
            name="location"
            value={locationField}
            placeholder="Search Location"
            className={styles.input}
            onChange={(event) => {
              const value = event.target.value;
              dispatch(changeLocationField(value));
            }}
          />
          <LocationSuggestion />
          <button
            aria-label="search"
            className={styles.searchButton}
            type="submit"
          >
            <FaSearch size={"1.5em"} color="grey" />
          </button>
        </div>
      </div>
    </form>
  );
}

export default SearchForm;
