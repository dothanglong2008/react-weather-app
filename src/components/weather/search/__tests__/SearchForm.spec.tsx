import renderer from "react-test-renderer";
import SearchForm from "../SearchForm";
import { Provider } from "react-redux";
import { store } from "../../../../app/store";

describe("Test SearchForm component", () => {
  it("renders SearchForm correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SearchForm />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
