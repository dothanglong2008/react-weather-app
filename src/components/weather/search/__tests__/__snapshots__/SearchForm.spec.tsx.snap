// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchForm component renders SearchForm correctly 1`] = `
<form
  className="form"
  name="search"
  onSubmit={[Function]}
>
  <div
    className="searchContainer"
  >
    <div
      className="inputContainer"
    >
      <input
        className="input"
        id="location"
        name="location"
        onChange={[Function]}
        placeholder="Search Location"
        type="text"
        value=""
      />
      <button
        className="suggestion"
        onClick={[Function]}
        title="Select current location"
        type="button"
      >
        <svg
          color="grey"
          fill="currentColor"
          height="2em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "grey",
            }
          }
          viewBox="0 0 24 24"
          width="2em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <path
            d="M13 4.069V2h-2v2.069A8.01 8.01 0 0 0 4.069 11H2v2h2.069A8.008 8.008 0 0 0 11 19.931V22h2v-2.069A8.007 8.007 0 0 0 19.931 13H22v-2h-2.069A8.008 8.008 0 0 0 13 4.069zM12 18c-3.309 0-6-2.691-6-6s2.691-6 6-6 6 2.691 6 6-2.691 6-6 6z"
          />
        </svg>
      </button>
      <button
        aria-label="search"
        className="searchButton"
        type="submit"
      >
        <svg
          color="grey"
          fill="currentColor"
          height="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "grey",
            }
          }
          viewBox="0 0 512 512"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </button>
    </div>
  </div>
</form>
`;
